src/product_api/interface.rs

```diff
@@ -6,8 +6,106 @@ use serde_with::{formats::PreferOne, serde_as, DefaultOnError, OneOrMany};
 
 use crate::interface::{AgeCategory, FileType, WorkCategory, WorkType};
 
+pub type StrOrBool = TwoPossibility<String, bool>;
+pub type ArrOrSingle<T> = TwoPossibility<Vec<T>, T>;
+pub type HashMapOrArr<T> = TwoPossibility<HashMap<String, T>, Vec<T>>;
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(untagged)]
+pub enum TwoPossibility<T, V> {
+    One(T),
+    Two(V),
+}
```

Comment:
I think `either` crate can be used instead of defining own type but this maybe enough for now.